File: USACO/Problems/2023 DEC Bronze #1 - Candy Cane Feast/23_Dec_Bronze_1_candyCaneFeast.cpp
// 만점은 아님, 수정 필요

#include <iostream>
#include <vector>

using namespace std;
using ll = long long;

int main(){
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int n, M;
    cin >> n >> M;

    // 입력, 분류 저장
    vector<ll> cowHeights(n);
    for(int i = 0; i < n; i++){
        cin >> cowHeights[i];
    }

    vector<ll> candyHeights(M);
    for(int i = 0; i < M; i++){
        cin >> candyHeights[i];
    }

    for (int caneN = 0; caneN < M; caneN++){
        ll candyBottom = 0;
        ll candyTop = candyHeights[caneN];
        for (int cowN = 0; cowN < n; cowN++){
            if (cowHeights[cowN] > candyBottom){
                ll eaten = min((ll)cowHeights[cowN], candyTop) - candyBottom;
                cowHeights[cowN] += eaten;
                candyBottom += eaten;
            }
        }
    }

    for (ll cowHeight : cowHeights){
        cout << cowHeight << "\n";
    }
}

File: USACO/Problems/2023 DEC Bronze #1 - Candy Cane Feast/23_Dec_Bronze_1_candyCaneFeast.py
n, M = list(map(int, input().split()))
cowHeights = list(map(int, input().split()))
candyHeights = list(map(int, input().split()))

for caneN in range(M):
    candyBottom = 0
    candyTop = candyHeights[caneN]
    for cowN in range(n):
        # 소가 캔디 케인 하단에 닿을 수 있으면 먹음
        if cowHeights[cowN] > candyBottom:
            # 먹을 수 있는 양 = min(소의 높이, 캔디 상단) - 캔디 하단
            eaten = min(cowHeights[cowN], candyTop) - candyBottom
            cowHeights[cowN] += eaten
            candyBottom += eaten

for cowHeight in cowHeights:
    print(cowHeight)